consensus/polybft/polybft.go

```diff
@@ -601,10 +601,11 @@ func (p *Polybft) startConsensusProtocol() {
 		stopSequence func()
 	)
 
+	// check 2 seconds above block time, so we don't artificially close a sequence before the actual sequence has ended properly
+	checkDuration := time.Duration(p.config.BlockTime + 2)
+	staleChecker := newStaleSequenceCheck(p.logger, p.blockchain.CurrentHeader, checkDuration*time.Second)
```

Comment:
Just thinking out loud, would calculating offset as blockTime/2 make more sense, instead of hardcoding it to 2 secs? Though otoh, for Palm it probably makes sense hardcoding it this way as well.

```suggestion
	offset := p.config.BlockTime
	if offset > 1 {
		offset = offset / 2
	}

	checkDuration := time.Duration(p.config.BlockTime + offset)
```

Reply:
I went back and forth on this one and settled on a + 2 second.  Not for Palm, but because we're expecting a block every X seconds, and for it to be confirmed in less than 2 seconds regardless of block time.  But it does make sense to make this change to avoid unnecessary checks.  Even with +2 seconds I do see the check appearing in the logs.  I'll make the change 👍